### PR TITLE
OTG Test - TE-8.1: DUT Daemon Failure

### LINKED ISSUE #894: OTG: TE-8.1: DUT Daemon Failure

### PR DESCRIPTION
**Pull request by octpetre**
OTG Test conversion of **_TE-8.1: DUT Daemon Failure_**
Fixes #894 
Test log output [here](https://drive.google.com/file/d/1XbB1uVzzVzdw0YjA0w8ACPY1WhbEZIz4/view?usp=share_link)

### CODE REVIEW

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+
+	p1 := dut.Port(t, "port1")
+	i1 := &oc.Interface{Name: ygot.String(p1.Name())}
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), configInterfaceDUT(i1, &dutPort1))
+
+	p2 := dut.Port(t, "port2")
+	i2 := &oc.Interface{Name: ygot.String(p2.Name())}
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), configInterfaceDUT(i2, &dutPort2))
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:125]: Please copy the ExplicitInterfaceinDefaultVRF and ExplicitPortSpeed deviations here

(Added to the ATE test in:
- https://github.com/openconfig/featureprofiles/commit/656ff4967fef66a0306e8fa8d416b570b9aafc9e#diff-54a6920d26a5dc282fb7809685af54ac4425e025a55084a0c793774875981186)
- https://github.com/openconfig/featureprofiles/commit/71695eafc3b6cdf0625b4a42b98f093d41b52d47#diff-54a6920d26a5dc282fb7809685af54ac4425e025a55084a0c793774875981186
)

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+		Name:    "atePort2",
+		MAC:     "02:00:02:01:01:01",
+		IPv4:    "192.0.2.6",
+		IPv4Len: ipv4PrefixLen,
+	}
+
+	gRIBIDaemons = map[ondatra.Vendor]string{
+		ondatra.ARISTA:  "Gribi",
+		ondatra.CISCO:   "emsd",
+		ondatra.JUNIPER: "rpd",
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:92]: Please add the NOKIA mapping that's now in the ATE version: https://github.com/openconfig/featureprofiles/commit/4e64df8a3ccd4f09b18c683836383f7c400c34f3

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+		s4.Enabled = ygot.Bool(true)
+	}
+	s4a := s4.GetOrCreateAddress(a.IPv4)
+	s4a.PrefixLength = ygot.Uint8(ipv4PrefixLen)
+
+	return i
+}
+
+// configureDUT configures port1 and port2 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:116]: t.Helper

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+	i1 := &oc.Interface{Name: ygot.String(p1.Name())}
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), configInterfaceDUT(i1, &dutPort1))
+
+	p2 := dut.Port(t, "port2")
+	i2 := &oc.Interface{Name: ygot.String(p2.Name())}
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), configInterfaceDUT(i2, &dutPort2))
+}
+
+// configureATE configures port1 and port2 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) gosnappi.Config {
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:129]: t.Helper

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+	otg := ate.OTG()
+	top := otg.NewConfig(t)
+
+	p1 := ate.Port(t, "port1")
+	p2 := ate.Port(t, "port2")
+
+	atePort1.AddToOTG(top, p1, &dutPort1)
+	atePort2.AddToOTG(top, p2, &dutPort2)
+
+	flow := top.Flows().Add().SetName("Flow")
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:139]: Move the flow name to a constant that can be used in all functions.

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go (hunk 1 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:125]: Deviations from ATE test added

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go (hunk 2 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:92]: mapping added from the ATE test

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go (hunk 3 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:116]: Done

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go (hunk 4 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:129]: Done

File: feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go (hunk 5 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/dut_daemon_failure/dut_daemon_failure_test.go:139]: Done

**Review by jasdeep-hundal — APPROVED** [review]

**Pull request merged**